EIPS/eip-rpc.md

```diff
@@ -0,0 +1,2314 @@
+
+<details>
+<summary><code><strong>eth_submitHashrate</strong></code></summary>
+
+#### Description
+
+Submit a mining hashrate
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Data`](#data)}|hash rate|
+|2|{[`Data`](#data)}|random ID identifying this node|
+
+#### Returns
+
+{`boolean`} - `true` if submitting went through successfully, `false` otherwise
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_submitHashrate",
+    "params": [
+        "0x0000000000000000000000000000000000000000000000000000000000500000",
+        "0x59daa26581d0acd1fce254fb7e85952f4c09d0915afd33d3886cd914bc7d283c"
+    ]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": true
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_submitWork</strong></code></summary>
+
+#### Description
+
+Submit a proof-of-work solution
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Data`](#data)}|nonce found|
+|2|{[`Data`](#data)}|header's pow-hash|
+|3|{[`Data`](#data)}|mix digest|
+
+#### Returns
+
+{`boolean`} - `true` if the provided solution is valid, `false` otherwise
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_submitWork",
+    "params": [
+        "0x0000000000000001",
+        "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef",
+        "0xD1GE5700000000000000000000000000D1GE5700000000000000000000000000"
+    ]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": true
+}
+```
+---
+</details>
+
+
+<details>
+<summary><code><strong>eth_syncing</strong></code></summary>
+
+#### Description
+
+Returns information about the status of this client's network synchronization
+
+#### Parameters
+
+_(none)_
+
+#### Returns
+
+{`boolean|object`} - `false` if this client is not syncing with the network, otherwise an object with the following members:
+
+- {[`Quantity`](#quantity)} `currentBlock` - number of the most-recent block synced
+- {[`Quantity`](#quantity)} `highestBlock` - number of latest block on the network
+- {[`Quantity`](#quantity)} `startingBlock` - block number at which syncing started
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_syncing",
+    "params": []
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": {
+        "currentBlock": '0x386',
+        "highestBlock": '0x454',
+        "startingBlock": '0x384'
+    }
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_uninstallFilter</strong></code></summary>
+
+#### Description
+
+Destroys a filter based on filter ID
+
+**Note:** This should only be called if a filter and its notifications are no longer needed. This will also be called automatically on a filter if its notifications are not retrieved using `eth_getFilterChanges` for a period of time.
```

Comment:
It seems beneficial to provide a *recommended* timeout length here so that different clients are more likely to use similar timeouts.